### DISCUSSION TITLE
Frontmatter stripped

### DISCUSSION THREAD

**Discussion by pbougie** [tinacms/tinacms#4307] [Help] [answered]
I have internal frontmatter in my Markdown files that keeps getting stripped when Tina saves a file. How can I force Tina to keep the extra frontmatter? Is there a way to specify it without showing it in the editor? Thanks.

**Comment by kldavis4** [comment]
Can you provide more information? Is it all frontmatter or just some of the fields? Do you have a reproduction?

**Reply by pbougie** [reply]
@kldavis4 I have the following frontmatter in my Markdown file:

```
---
title: Title Goes Here
description: Description goes here...
layout: common
---
...
```

I've made both title and description editable in Tina. The `layout` attribute is internal to my static site generator and I don't want it to be editable in Tina. However when the Markdown file is saved, Tina strips the attribute out of the file and breaks my build. Why is Tina removing unknown attributes? Or is there a way to specify these cases in the Tina config to retain the attributes without showing them in the editor? Thanks.

**Reply by kldavis4** [reply, marked as answer]
define `layout` in your schema and set the following on the field definition to hide it:

```
ui:{
   component: 'hidden',
}
```

**Reply by pbougie** [reply]
@kldavis4 Thanks, your solution worked!
